src/ddscxx/include/org/eclipse/cyclonedds/topic/BuiltinTopicTraits.hpp

```diff
@@ -16,6 +16,142 @@
 #include <org/eclipse/cyclonedds/topic/TopicTraits.hpp>
 #include <org/eclipse/cyclonedds/topic/BuiltinTopicCopy.hpp>
 
-/* TODO: add builtin topic traits. */
+namespace org
+{
+namespace eclipse
+{
+namespace cyclonedds
+{
+namespace topic
+{
+
+template <>
+class TopicTraits<dds::topic::ParticipantBuiltinTopicData>
+{
+public:
+  static bool isKeyless()
+  {
+    return true;
+  }
+
+  static const char *getTypeName()
+  {
+    return "dds::topic::ParticipantBuiltinTopicData";
+  }
+
+  static ddsi_sertype *getSerType()
+  {
+    auto *st = new ddscxx_sertype<dds::topic::ParticipantBuiltinTopicData>();
+    return static_cast<ddsi_sertype*>(st);
+  }
+
+  static size_t getSampleSize()
+  {
+    return sizeof(dds::topic::ParticipantBuiltinTopicData);
+  }
+
+  static bool isSelfContained()
+  {
+    return false;
```

Comment:
Why is the ParticipantBuiltinTopicData type not considered selfContained? Is that because it has an unbounded sequence which forces it to have an indirection to a separately allocated part? How exactly do you define the term SelfContained in this context? Is this only relevant in the context of the IceOryx integration, where only SelfContained types may be communicated through shared memory?
(The same questions also applies to the other Builtin topic types, but I am not going to repeat it for every one.)